src/main/java/com/toptrumps/cli/CommandLineUI.java

```diff
@@ -49,4 +57,30 @@ public void startGame() {
         System.out.println("Thanks for playing!");
     }
 
-}
+
+    public void askHowManyPlayers() {
+        try {
+
+            System.out.println("How many players do you want to play against?");
+            System.out.println("Please select 1-4");
+
+            int players = scanner.nextInt();
+            while (players < MIN_PLAYERS || MAX_PLAYERS > 4) {
```

Comment:
Is this line should be like this:
`while (players < MIN_PLAYERS || players > MAX_PLAYERS) {`
?